pkg/linter/linter_test.go

```diff
@@ -72,4 +72,105 @@ func TestLinter_Lint(t *testing.T) {
 		require.Len(t, results, 0, "linter should ignore mixins that doesn't support the lint command")
 	})
 
+	t.Run("does not use reserved prefix", func(t *testing.T) {
+		cxt := portercontext.NewTestContext(t)
+		mixins := mixin.NewTestMixinProvider()
+		parameterName := "porter-debug"
+		l := New(cxt.Context, mixins)
+		param := map[string]manifest.ParameterDefinition {
+			"A": {
+				Name: parameterName,
+			},
+		}
+
+		m := &manifest.Manifest{
+			Parameters: param,
+		}
+		mixins.LintResults = Results{
+			{
+				Level: LevelWarning,
+				Location: Location {
+					Action: "",
+					Mixin: "",
+					StepNumber: 0,
+					StepDescription: "",
+			
+				},
+				Code: "exec-100",
+				Title: "Reserved name warning",
+				Message: parameterName + " has a reserved prefix",
+				URL: "",
+			},
+		}
+
+		results, err := l.Lint(ctx, m)
+		require.NoError(t, err, "Lint failed")
+		require.Len(t, results, 1, "linter should have returned 1 result")
+		require.Equal(t, mixins.LintResults, results, "unexpected lint results")
+	})
+
+	t.Run("is case insensitive and does not use reserved prefix even if mixed case or upper", func(t *testing.T) {
```

Comment:
You do not need to change your tests here unless you want to try it out; Your tests work fine. But I'd like to share a way to have the same number of test cases with a bit less copy/paste code.

Here's an example of how to create testcases or a test matrix. The testcase contains unique inputs for the testcase, such as the parameter name and if we expect an error or not.

Then the code creates a new subtest for each test cases and uses the testcase data structure to validate that test case.

https://github.com/getporter/porter/blob/138f83d0a2568feb5ff279724ed543f1fd0bbe6f/pkg/porter/parameters_test.go#L456

If you search the codebase for `testcases :=` you will find a lot more examples that you can use to get familiar with this way of writing tests in Go.